utils/analytics.ts

```diff
@@ -0,0 +1,68 @@
+import {
+  getAnalytics,
+  getAppInstanceId,
+  logEvent,
+  setAnalyticsCollectionEnabled,
+  logScreenView,
+} from '@react-native-firebase/analytics';
+import { getApp } from '@react-native-firebase/app';
+
+const app = getApp();
+const analytics = getAnalytics(app);
+
+const allowTracking = async () => {
+  const appInstanceId = await getAppInstanceId(analytics);
+  if (!appInstanceId) {
+    await setAnalyticsCollectionEnabled(analytics, true);
+  }
+};
+const trackEvent = (category: string, action: string, label: string) => {
+  logEvent(analytics, category, { [action]: label });
```

Comment:
Use a try-catch block, when tracking events. If firebase analytics fails, the app could crash or events could be lost silently.

Reply:
If the category, action or label sent to this function is invalid or undefined, there's no sanitization, it will push to analytics with undefined value.